### ISSUE TITLE
Remote BrowserWindow APIs with callback don't work

### ISSUE THREAD

**Issue by rhysd** [electron/spectron#46]
Hi, I'm considering to take a screenshot of app when test failed.  I found `BrowserWindow`'s `capturePage` method but it doesn't work from Spectron as below.

### 2. `$ node below-code.js`

```javascript
'use strict';
const Application = require('spectron').Application;
const electron = require('electron-prebuilt');

const app = new Application({path: electron});
app.start();

// Wait for app starting
setTimeout(() => {
    console.log('start');
    app.browserWindow
        .capturePage(img => {
            console.log(img);
            const fs = require('fs');
            fs.writeFile('screenshot.png', img.toPng());
        })
        .then(result => {
            console.log('RESULT:', result);
            app.stop();
        });
}, 3000);
```

## Environment

- Spectron 3.1.0
- Electron 1.1.3
- OS X 10.11.4 (MacBook Pro)

**Comment by kevinsawicki** [comment]
Thanks for raising this issue, a fix has been published in `3.1.1`.

Since all spectron APIs return promises, you won't specify the callback as an argument to `capturePage` but instead call `then` on the return value and the promise should be resolved with a node `Buffer` containing the raw image data. It can't return a `nativeImage` since those aren't JSON serializable and are only available in Electron apps since it is a Chrome-specific API.

You can write it directly out as a png though:

```js
app.browserWindow.capturePage().then(function (imageBuffer) {
  fs.writeFile('page.png', imageBuffer)
})
```

And it does support the `rect` parameter if you only want to get a part of the page:

```js
app.browserWindow.capturePage({x: 0, y: 0, width: 10, height: 10}).then(function (imageBuffer) {
  fs.writeFile('page.png', imageBuffer)
})

**Comment by rhysd** [comment]
Thank you!  It's very helpful.  I'll confirm the fix.
